Lib/test/test_warnings/__init__.py

```diff
@@ -149,9 +149,9 @@ def test_always(self):
             self.module.filterwarnings("always", category=UserWarning)
             message = "FilterTests.test_always"
             self.module.warn(message, UserWarning)
-            self.assertTrue(message, w[-1].message)
+            self.assertEqual(message, w[-1].message.args[0])
```

Comment:
There are other flaws in this test, due to which the second assertion doesn't have effect. Rewrite the test as:
```python
    def test_always(self):
        with original_warnings.catch_warnings(record=True,
                module=self.module) as w:
            self.module.resetwarnings()
            self.module.filterwarnings("always", category=UserWarning)
            message = "FilterTests.test_always"
            def f():
                self.module.warn(message, UserWarning)
            f()
            self.assertEqual(len(w), 1)
            self.assertEqual(w[-1].message.args[0], message)
            f()
            self.assertEqual(len(w), 2)
            self.assertEqual(w[-1].message.args[0], message)
```